new/language/implementation/javascript/javascript.go

```diff
@@ -0,0 +1,240 @@
+	}
+
+	return replaced, params, nil
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) ExtractPatternMatchNode(input string) (string, int, error) {
+	inputBytes := []byte(input)
+	matches := matchNodeRegex.FindAllIndex(inputBytes, -1)
+
+	if len(matches) == 0 {
+		return input, 0, nil
+	}
+
+	if len(matches) > 1 {
+		return "", 0, errors.New("pattern must only contain a single match node")
+	}
+
+	match := matches[0]
+	return string(inputBytes[0:match[0]]) + string(inputBytes[match[1]:]), match[0], nil
+}
+
+// TODO: See if anything needs to be added here
+func produceDummyValue(i int, nodeType string) string {
+	switch nodeType {
+	case "identifier", "call":
+		return "curioVar" + fmt.Sprint(i)
+	default:
+		return "CurioVar" + fmt.Sprint(i)
+	}
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) AnonymousPatternNodeParentTypes() []string {
+	return anonymousPatternNodeParentTypes
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) FindPatternMatchNode(input []byte) [][]int {
+	return matchNodeRegex.FindAllIndex(input, -1)
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) FindPatternUnanchoredPoints(input []byte) [][]int {
+	return ellipsisRegex.FindAllIndex(input, -1)
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) IsTerminalDetectionNode(node *tree.Node) bool {
+	return false
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) PatternMatchNodeContainerTypes() []string {
+	return patternMatchNodeContainerTypes
+}
+
+func (implementation *javascriptImplementation) PatternIsAnchored(node *tree.Node) bool {
+	// FIXME: implement this
+	parent := node.Parent()
+	if parent == nil {
+		return true
+	}
+
+	// Class body
+	if parent.Type() == "class_declaration" && !node.Equal(parent.ChildByFieldName("name")) {
+		return false
+	}
+
+	// arrow functions
+	if parent.Type() == "arrow_functions" && !node.Equal(parent.ChildByFieldName("parameters")) {
+		return false
+	}
+
+	// function
+	if parent.Type() == "function_declaration" && !node.Equal(parent.ChildByFieldName("name")) && !node.Equal(parent.ChildByFieldName("parameters")) {
+		return false
+	}
+
+	// method
+	if parent.Type() == "method_definition" && !node.Equal(parent.ChildByFieldName("name")) && !node.Equal(parent.ChildByFieldName("parameters")) {
+		return false
+	}
+
+	return true
+}
+
+// TODO: See if anything needs to be added here
+func (implementation *javascriptImplementation) DescendIntoDetectionNode(node *tree.Node) bool {
+	// parent := node.Parent()
+
+	// if parent != nil && parent.Type() == "member_expression" && node.Equal(parent.ChildByFieldName("object")) {
+	// 	return slice.Contains(passThroughMethods, parent.ChildByFieldName("method").Content())
+	// }
```

Comment:
We probably want something like this:
```suggestion
	parent := node.Parent()

	if parent != nil && parent.Type() == "member_expression" && node.Equal(parent.ChildByFieldName("object")) {
	 	return false
	}
```